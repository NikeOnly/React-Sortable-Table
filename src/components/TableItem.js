import React from 'react';

export default class TableItem extends React.PureComponent {
    onDeleteField() {
      this.props.deleteField(this.props.field)
    }

    render() {
        const {field} = this.props;
        return (
            <tbody>
                <tr>
                    <td>{field.id}</td>
                    <td>{field.firstName}</td>
                    <td>{field.lastName}</td>
                    <td>{field.phone}</td>
                    <td>{field.gender}</td>
                    <td>{field.age}</td>
                    <td><button className="btn btn-primary btn-sm" onClick={this.onDeleteField.bind(this)}>Delete</button></td>
                </tr>
            </tbody>
        );
    }
}
