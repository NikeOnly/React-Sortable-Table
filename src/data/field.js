import generateUuid from 'uuid/v1';

export class Field {
    constructor(id, firstName, lastName, phone, gender, age) {
        this.code = generateUuid();
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.phone = phone;
        this.gender = gender;
        this.age = age;
    }
}
