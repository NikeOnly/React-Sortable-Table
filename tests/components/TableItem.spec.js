import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';

import TableItem from '../../src/components/TableItem';
import {Field} from '../../src/data/field';

const field = new Field(1, 'Melisa', 'Black', '11111111', 'female', '28' )

describe('TableItem', () => {
    test('match snapshot', () => {
        const tree = renderer.create(<TableItem field={field} />).toJSON()
        expect(tree).toMatchSnapshot();
    })

    test('click delete button', () => {        
        let deleteField = jest.fn()

        const wrapper = shallow(<TableItem field={field} deleteField={deleteField} />)
        const button = wrapper.find('button')

        button.simulate('click')

        expect(deleteField).toHaveBeenCalledWith(field)
    })
})
