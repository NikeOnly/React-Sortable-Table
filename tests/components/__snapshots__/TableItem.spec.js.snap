// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableItem match snapshot 1`] = `
<tbody>
  <tr>
    <td>
      1
    </td>
    <td>
      Melisa
    </td>
    <td>
      Black
    </td>
    <td>
      11111111
    </td>
    <td>
      female
    </td>
    <td>
      28
    </td>
    <td>
      <button
        className="btn btn-primary btn-sm"
        onClick={[Function]}
      >
        Delete
      </button>
    </td>
  </tr>
</tbody>
`;
